source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -737,10 +737,6 @@ ServerContextImpl::ServerContextImpl(Stats::Scope& scope,
       if (!response->matchesCertificate(*tls_context_.cert_contexts_[i].cert_chain_)) {
         throw EnvoyException("OCSP response does not match its TLS certificate");
       }
-      if (response->isExpired()) {
```

Comment:
Did we verify upstream also stripped this? Or is this specific to us?

Reply:
This is specific to us and the decision to remove it was made in discussion with @twghu

Reply:
great, thanks for sharing context. perhaps leave note for our future selves in the code? e.g. something like:

```
// MAISTRA - removed "if (response->isExpired()" handling, because .. 
```

Reply:
Reinstated the check

Reply:
The expiry check if() was added 
on 13/4/21 https://github.com/maistra/envoy/commit/412c091b302cc688bf3269022bf679bced828b2f#diff-2b39f212c2f760ae547351a46f92ad08c0ec84c556d678ebe9388b9d008e7313

and was removed on 20/5/21
https://github.com/maistra/envoy/commit/963afc8094dc2c17bd24526f2de11841a5904b1b

The last commit should indicate reason for removal, a sanity check would be a good idea.

Reply:
So digging into this, it seems the check is (also) done in [ServerContextImpl::ocspStapleAction](https://github.com/maistra/envoy/blob/0428980c9210b2c7de98eb64e73fcfc526c96426/source/extensions/transport_sockets/tls/context_impl.cc#L948).  So I think we can remove it.